common/app/assets/assets.scala

```diff
@@ -29,27 +29,40 @@ class AssetMap(base: String, assetMap: String) {
   }
 
   private def assets(): Map[String, Asset] = {
-
-    // Use the grunt-generated asset map in Dev.
-    val json: String = if (Play.current.mode == Mode.Dev) {
-      val assetMapUri = new java.io.File(s"static/hash/" + assetMap).toURI
-      IOUtils.toString(assetMapUri)
-    } else {
-      val url = AssetFinder(assetMap)
-      IOUtils.toString(url)
-    }
-    val js: JsObject = Json.parse(json).asInstanceOf[JsObject]
-
-    val paths = js.fields.toMap mapValues { _.asInstanceOf[JsString].value }
-
     paths mapValues { path => Asset(base + path) }
   }
 
   private lazy val memoizedAssets = assets()
 }
 
+class JspmAssets(base: String) {
+  val jsonString: String = IOUtils.toString(AssetFinder("assets/systemjs-bundle-config.json"))
+  val paths: Map[String, String] = Json.parse(jsonString).validate[Map[String, List[String]]] match {
+    case JsSuccess(myMap, _) => myMap
+      .mapValues(_.mkString)
+      .map(_.swap)
+      .map{case (k, v) => (s"javascripts/$k.js", s"$v.js")}
+    case JsError(errors) => Map.empty
+  }
+
+  val lookup = new AssetMap(base, paths)
+  def apply(path: String): Asset = lookup(path)
+}
```

Comment:
The jspm asset map is in format `Map[BundleFilePath, List[BundledModuleFilePath]]` when we need `Map[BundledModuleFilePath, BundleFilePath]`, hence the munging.